### PR TITLE
Create report form integrated with maps page

### PR DESCRIPTION
**Pull request by sandlerben**

### CODE REVIEW

**Comment by sandlerben** [comment]
@kongsally make sure that all the tests pass

**Comment by sandlerben** [comment]
Can we improve the styling of this a bit? https://www.dropbox.com/s/t1hty489y31ul4l/Screenshot%202016-01-21%2014.37.18.png?dl=0
- There is a box within a box
- The text box for 'Additional Notes' goes beyond the box
- Use current location is a blue "action" button while submit is an ordinary button. Should this be reversed?

File: IncidentReports-2016-01-16.csv [hunk 1]
```diff
@@ -0,0 +1,202 @@
+DATE,LOCATION,AGENCY ID,VEHICLE ID,DURATION,LICENSE PLATE,DESCRIPTION
```
Comment (sandlerben) [inline IncidentReports-2016-01-16.csv]: What is this file?

File: app/admin/views.py [hunk 2]
```diff
@@ -1,9 +1,9 @@
 from ..decorators import admin_required
-
-from flask import render_template, abort, redirect, flash, url_for, request
+from flask import render_template, abort, redirect, flash, url_for, \
```
Comment (sandlerben) [inline app/admin/views.py]: Can you change to

``` python
from flask import (
   <one>,
   <two>,
   ...
)
```

File: app/admin/views.py [hunk 3]
```diff
@@ -15,10 +15,11 @@
     AddAgencyForm,
 )
 from . import admin
-from ..models import User, Role, Agency, EditableHTML
+from ..models import User, Role, Agency, EditableHTML, IncidentReport
 from .. import db
 from ..utils import parse_phone_number
 from ..email import send_email
+import csv
```
Comment (sandlerben) [inline app/admin/views.py]: Group all python standard library imports (datetime, csv) together probably at the top

File: app/admin/views.py [hunk 4]
```diff
@@ -314,3 +315,31 @@ def update_editor_contents():
     db.session.commit()
 
     return 'OK', 200
+
+
+@admin.route('/download_reports', methods=['GET'])
```
Comment (sandlerben) [inline app/admin/views.py]: I'm going to check out this change and put it on its own branch. Ok?

File: app/main/forms.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
+from flask.ext.wtf import Form
+from wtforms.fields import StringField, SubmitField, IntegerField, TextAreaField, HiddenField, \
```
Comment (sandlerben) [inline app/main/forms.py]: Change to

``` python
from wtforms.fields import (
   <one>,
   ...
)
```

File: app/main/forms.py [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+from wtforms.validators import (
+    InputRequired,
+    Length,
+    Optional,
+    NumberRange,
+    URL
+)
+from app.custom_validators import StrippedLength
+from .. import db
+import datetime as datetime
```
Comment (sandlerben) [inline app/main/forms.py]: The `as datetime` is redundant

Comment (sandlerben) [inline app/main/forms.py]: Also see comment above about grouping imports a bit more logically

File: app/main/forms.py [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+
+class IncidentReportForm(Form):
```
Comment (sandlerben) [inline app/main/forms.py:25]: Uh oh, this should be in `reports/forms.py`. Did you move it back here?

File: app/main/forms.py [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+
+    latitude = HiddenField('Latitude')
+    longitude = HiddenField('Longitude')
+    location = StringField('Address')
+    today = datetime.date.today()
+    date = DateField('Date', default=today,
+                     validators=[Optional()])
+
+    # TODO - add support for h:m:s format
+    duration = IntegerField('Idling Duration (minutes)', validators=[
```
Comment (sandlerben) [inline app/main/forms.py:62]: This is out of date. I think recent changes weren't incorporated because all changes have been to `reports/forms.py`

File: app/main/views.py [hunk 9]
```diff
@@ -1,15 +1,37 @@
 from flask import render_template
-from ..models import EditableHTML, IncidentReport
+from ..models import EditableHTML
```
Comment (sandlerben) [inline app/main/views.py]: Note that there is a `from app.models` import below. You should from all the models imports together.

File: app/models/incident_report.py [hunk 10]
```diff
@@ -128,7 +128,6 @@ def rand_alphanumeric(n):
                 picture_url=fake.image_url(),
                 description=fake.paragraph(),
                 notify_workers_upon_creation=False,
-                **kwargs
```
Comment (sandlerben) [inline app/models/incident_report.py:134]: Why did you remove this?

File: app/static/js/initialize_map.js [hunk 11]
```diff
@@ -0,0 +1,123 @@
+geocoder = new google.maps.Geocoder();
```
Comment (sandlerben) [inline app/static/js/initialize_map.js:1]: Why is this showing up as an addition? Wasn't this all here?

File: mydump.csv [hunk 12]
```diff
@@ -0,0 +1,101 @@
+VEHICLE ID,LICENSE PLATE,LOCATION,DATE,DURATION,AGENCY ID,DESCRIPTION
```
Comment (sandlerben) [inline mydump.csv]: No reason for this or `dump.csv` to be in git

File: app/main/forms.py (hunk 7 above)
Comment (kongsally) [inline app/main/forms.py:25]: I think I merged master's and also have reports/forms.py. I guess I'll just remove this file then?

**Comment by sandlerben** [comment]
Actually, I wouldn't delete poll244.csv since we need it in git to get it up to heroku

**Comment by kongsally** [comment]
@sandlerben 
Edited Incident Report's styling for the create report button and text area,
Edited the import statement for views.py,
Added back kwargs (not sure why I deleted it first... Maybe I had issues with the outdated forms model I was using)

File: app/static/js/initialize_map.js (hunk 11 above)
Comment (kongsally) [inline app/static/js/initialize_map.js:1]: Not sure why this happened.
